core/domain/suggestion_services.py

```diff
@@ -2073,42 +2098,45 @@ def _update_translation_review_stats_models(translation_review_stats):
             stat.accepted_translations_count)
         stats_model.accepted_translations_with_reviewer_edits_count = (
             stat.accepted_translations_with_reviewer_edits_count)
-        stats_model.accepted_translation_word_count = (
-            stat.accepted_translation_word_count)
         stats_model.first_contribution_date = (
             stat.first_contribution_date)
         stats_model.last_contribution_date = (
             stat.last_contribution_date)
 
-    suggestion_models.TranslationReviewStatsModel.update_timestamps_multi(
+    suggestion_models.TranslationReviewStatsModel.update_timestamps_multi(  # type: ignore[type-var]
         stats_models_to_update,
         update_last_updated_time=True)
-    suggestion_models.TranslationReviewStatsModel.put_multi(
+    suggestion_models.TranslationReviewStatsModel.put_multi(  # type: ignore[type-var]
         stats_models_to_update)
 
 
-def _update_question_contribution_stats_models(question_contribution_stats):
+def _update_question_contribution_stats_models(
+    question_contribution_stats: List[
+        suggestion_registry.QuestionContributionStats]) -> None:
     """Updates QuestionContributionStatsModel models for given question
     contribution stats.
 
     Args:
-        question_contribution_stats: list(QuestionContribution). A list of
-        QuestionContribution domain objects.
+        question_contribution_stats: list(QuestionContributionStats). A list of
+            QuestionContribution domain objects.
     """
     stats_ids = []
+    stats_dict = {}
 
     for stat in question_contribution_stats:
-        stats_ids.append(
-            suggestion_models.QuestionContributionStatsModel.generate_id(
-                stat.contributor_user_id,
-                stat.topic_id
-            ))
+        stat_id = suggestion_models.QuestionContributionStatsModel.build_id(
+            stat.contributor_user_id, stat.topic_id)
+        stats_ids.append(stat_id)
+        stats_dict[stat_id] = stat
 
     stats_models_to_update = (
         suggestion_models.QuestionContributionStatsModel.get_multi(stats_ids))
 
-    for index, stats_model in enumerate(stats_models_to_update):
-        stat = question_contribution_stats[index]
+    for stats_model in stats_models_to_update:
+
+         # Ruling out the possibility of None for mypy type checking.
```

Comment:
This and other similar comments need more detail. Please check with @sahiljoster32 for guidance on this.

Reply:
@chris7716 please put a proper reason why this is not None here.

But yeah if you are not sure then please raise an exception here, and after that you can use assert if it is still needed.

Reply:
Added the reasoning.